### PR TITLE
Implement `TriggerConfigModel`

### PR DESCRIPTION
**Pull request by alejoe91**
@dyf @jtyoung84 sorry this took a while on my side.

Implemented the `TriggerConfigModel` that we can use to unify calls to the aind-trigger-capsule. 

Note that the model accepts and parses strings with semicolon separation to attach multiple data assets and mount points. See after validation.

Let me know what you think!

### CODE REVIEW

**Comment by jtyoung84** [comment]
1) Should we put this in aind-trigger-capsule and import the model here instead?
2) I can wait until after this merged, but I can create a dev and main (prod) branch so people know which is running in production in aind-data-transfer-service

**Comment by alejoe91** [comment]
> 1. Should we put this in aind-trigger-capsule and import the model here instead?

Ideally this would be used also by the `aind-data-transfer-service` to validate the params to the aind-trigger-capsule, no?
Since the capsule is not pip-installable, I think here is a better fit!

**Comment by jtyoung84** [comment]
> > 1. Should we put this in aind-trigger-capsule and import the model here instead?
> 
> Ideally this would be used also by the `aind-data-transfer-service` to validate the params to the aind-trigger-capsule, no? Since the capsule is not pip-installable, I think here is a better fit!

It should be fine if we make things backwards compatible. If we need to introduce a breaking change, then it will require some coordination.

**Comment by alejoe91** [comment]
@jtyoung84 should be back-compatible. I ported all the parameters of the config currently passed to the trigger capsule: https://github.com/AllenNeuralDynamics/aind-data-transfer-models/blob/trigger-model/src/aind_data_transfer_models/trigger.py#L97-L118

Let me know if there are other actions to be taken here!

**Comment by jtyoung84** [comment]
@alejoe91 I pushed a small change in one of the tests to handle older versions of the `email_validator` package.

**Review by arielleleon — APPROVED** [review]
Looks good

**Review by jtyoung84 — APPROVED** [review]
We'll need to add a method to map the BasicUploadJobConfigs to the TriggerConfigModel in a future PR.

**Pull request merged**
